src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -6270,6 +6273,30 @@ bool Unit::isPossessing() const
         return false;
 }
 
+bool Unit::IsCharmerOrSelfPlayer() const
+{
+    if (IsCharmed())
+        return GetCharmerGUID().IsPlayer();
+    else
+        return IsPlayer();
+}
+
+Unit* Unit::GetCharmerOrSelf()
+{
+    if (IsCharmed())
+        return GetCharmer();
+    else
+        return this;
+}
+
+Player* Unit::GetCharmerOrSelfPlayer()
```

Comment:
`return ToPlayer(GetCharmerOrSelf())`, this can be inline in the header

Reply:
indeed, fixed